File: JvmClient/src/commonMain/kotlin/spaceEngineers/navigation/Node.kt
package spaceEngineers.navigation

import eu.iv4xr.framework.spatial.Vec3
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import spaceEngineers.graph.DataNode
import spaceEngineers.model.Vec3F

typealias NodeId = String
typealias Node = DataNode<NodeId, Vec3F>

File: Source/Ivxr.SpaceEngineers/Navigation/Node.cs
using Iv4xr.SpaceEngineers.WorldModel;

namespace Iv4xr.SpaceEngineers.Navigation
{
    public class Node
    {
        public readonly string Id;
        public readonly PlainVec3D Position;

        public Node(string id, PlainVec3D position)
        {
            Id = id;
            Position = position;
        }
    }
}
